fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -805,214 +843,377 @@ ACTOR Future<Void> releaseResolvingAfter(ProxyCommitData* self, Future<Void> rel
-	state Version prevVersion = versionReply.prevVersion;
+	self->commitVersion = versionReply.version;
+	self->prevVersion = versionReply.prevVersion;
 
 	for(auto it : versionReply.resolverChanges) {
-		auto rs = self->keyResolvers.modify(it.range);
+		auto rs = pProxyCommitData->keyResolvers.modify(it.range);
 		for(auto r = rs.begin(); r != rs.end(); ++r)
 			r->value().emplace_back(versionReply.resolverChangesVersion,it.dest);
 	}
 
-	//TraceEvent("ProxyGotVer", self->dbgid).detail("Commit", commitVersion).detail("Prev", prevVersion);
+	//TraceEvent("ProxyGotVer", pProxyContext->dbgid).detail("Commit", commitVersion).detail("Prev", prevVersion);
 
-	if (debugID.present())
-		g_traceBatch.addEvent("CommitDebug", debugID.get().first(), "MasterProxyServer.commitBatch.GotCommitVersion");
+	if (debugID.present()) {
+		g_traceBatch.addEvent(
+			"CommitDebug", debugID.get().first(),
+			"MasterProxyServer.commitBatch.GotCommitVersion"
+		);
+	}
 
-	ResolutionRequestBuilder requests( self, commitVersion, prevVersion, self->version, span );
+	return Void();
+}
+
+ACTOR Future<Void> getResolution(Context* self) {
+	ProxyCommitData* pProxyCommitData = self->pProxyCommitData;
+	std::vector<CommitTransactionRequest>& trs = self->trs;
+
+	ResolutionRequestBuilder requests(
+		pProxyCommitData,
+		self->commitVersion,
+		self->prevVersion,
+		pProxyCommitData->version,
+        self->span
+	);
 	int conflictRangeCount = 0;
-	state int64_t maxTransactionBytes = 0;
-	for (int t = 0; t<trs.size(); t++) {
+	self->maxTransactionBytes = 0;
+	for (int t = 0; t < trs.size(); t++) {
 		requests.addTransaction(trs[t].transaction, t);
-		conflictRangeCount += trs[t].transaction.read_conflict_ranges.size() + trs[t].transaction.write_conflict_ranges.size();
+		conflictRangeCount += (
+			trs[t].transaction.read_conflict_ranges.size() +
+			trs[t].transaction.write_conflict_ranges.size()
+		);
 		//TraceEvent("MPTransactionDump", self->dbgid).detail("Snapshot", trs[t].transaction.read_snapshot);
 		//for(auto& m : trs[t].transaction.mutations)
-		maxTransactionBytes = std::max<int64_t>(maxTransactionBytes, trs[t].transaction.expectedSize());
+		self->maxTransactionBytes = std::max<int64_t>(
+			self->maxTransactionBytes, trs[t].transaction.expectedSize()
+		);
 		//	TraceEvent("MPTransactionsDump", self->dbgid).detail("Mutation", m.toString());
 	}
-	self->stats.conflictRanges += conflictRangeCount;
+	pProxyCommitData->stats.conflictRanges += conflictRangeCount;
 
-	for (int r = 1; r<self->resolvers.size(); r++)
-		ASSERT(requests.requests[r].txnStateTransactions.size() == requests.requests[0].txnStateTransactions.size());
+	for (int r = 1; r < pProxyCommitData->resolvers.size(); r++)
+		ASSERT(requests.requests[r].txnStateTransactions.size() ==
+			requests.requests[0].txnStateTransactions.size());
 
-	// Sending these requests is the fuzzy border between phase 1 and phase 2; it could conceivably overlap with resolution processing but is still using CPU
```

Comment:
I feel we should keep this comments. Maybe move to the function's description?

Reply:
I think I will put it around the beginning of getResolution function